src/datasets/domain/models/Dataset.ts

```diff
@@ -0,0 +1,40 @@
+export interface Dataset {
+  id: number;
+  persistentId: string;
+  versionId: number;
+  versionInfo: DatasetVersionInfo;
+  license: DatasetLicense;
+  metadataBlocks: DatasetMetadataBlock[];
+}
+
+export interface DatasetVersionInfo {
+  majorNumber: number;
+  minorNumber: number;
+  state: DatasetVersionState;
+  createTime: Date;
+  lastUpdateTime: Date;
+  releaseTime?: Date;
+}
+
+export enum DatasetVersionState {
+  DRAFT = 'DRAFT',
+  RELEASED = 'RELEASED',
+  ARCHIVED = 'ARCHIVED',
+  DEACCESSIONED = 'DEACCESSIONED',
+}
+
+export interface DatasetLicense {
+  name: string;
+  uri: string;
+}
```

Comment:
In some cases there is also an image associated with the license
```suggestion
export interface DatasetLicense {
  name: string;
  uri: string;
  imageUrl?: string
}
```

![image](https://github.com/IQSS/dataverse-client-javascript/assets/23359572/7f6ab7a0-7116-4c50-8a9b-dae941d2119a)

Reply:
Good point. I've extended both the API endpoint and the package to include that.